Helpers/opent2t-translator-helper-wink/js/index.js

```diff
@@ -9,179 +9,184 @@ var apiEndpoint = 'https://api.wink.com';
+        // Start the request
+        request(options, function (error, response, body) {
+            if (!error && response.statusCode == 200) {
+                deferred.resolve(JSON.parse(body));
+            } else {
+                deferred.reject('Woops, there was an error putting device details: ' + error + ' (' + response.statusCode + ')');
+            }
+        });
 
-            console.log('Initialized Wink Helper.');
-        },
+        return deferred.promise;
+    }
 
-        setDesiredState: function (deviceType, deviceId, field, value) {
+    // Sets the desired state (of a single field)
+    setDesiredStateAsync(deviceType, deviceId, field, value) {
 
-            // q will help us with returning a promise
-            var deferred = q.defer();
+        // q will help us with returning a promise
+        var deferred = q.defer();
 
-            // build the object with desired state
-            var putBody = { 'data': { 'desired_state': {} } };
-            putBody.data.desired_state[field] = value;
+        // build the object with desired state
+        var putPayload = { 'data': { 'desired_state': {} } };
+        putPayload.data.desired_state[field] = value;
 
-            putDeviceDetails(deviceType, deviceId, putBody).then((response) => {
+        this.putDeviceDetailsAsync(deviceType, deviceId, putPayload).then((response) => {
 
-                // successfully put device details, parse out the desired state
-                // of the requested field in the response
-                var data = response.data;
-                if (!!data) {
-                    var desiredStateCollection = data['desired_state'];
+            // successfully put device details, parse out the desired state
+            // of the requested field in the response
+            var data = response.data;
+            if (!!data) {
+                var desiredStateCollection = data['desired_state'];
 
-                    if (!!desiredStateCollection) {
-                        deferred.resolve(desiredStateCollection[field]);
-                    } else {
-                        deferred.reject('Invalid response from server: no desired state collection.');
-                    }
+                if (!!desiredStateCollection) {
+                    deferred.resolve(desiredStateCollection[field]);
                 } else {
-                    deferred.reject('Invalid response from server: no data element.');
+                    deferred.reject('Invalid response from server: no desired state collection.');
                 }
-            }).catch((error) => {
-                // there was an error
-                deferred.reject(error);
-            });
-
-            return deferred.promise;
-        },
+            } else {
+                deferred.reject('Invalid response from server: no data element.');
+            }
+        }).catch((error) => {
+            // there was an error
+            deferred.reject(error);
```

Comment:
remove this set of code (add missing to putDeviceDetailsAsync) and just do return this.putDeviceDetailsAsync(....)